skyllh/core/analysis_utils.py

```diff
@@ -109,18 +109,27 @@ def polynomial_fit(
     ns : float
     """
     (params, cov) = np.polyfit(ns, p, deg, w=p_weight, cov=True)
+    if(deg == 2 and params[0] > 0):
```

Comment:
Please add a comment what this case is. It's not clear to me why you need to check for this